core/src/test/java/jenkins/plugins/DetachedPluginsUtilTest.java

```diff
@@ -27,22 +31,23 @@ public void checkJaxb() {
 
         assertEquals(new VersionNumber("11"), jaxb.getMinimumJavaVersion());
 
-        final List<DetachedPluginsUtil.DetachedPlugin> detachedPlugins = DetachedPluginsUtil.getDetachedPlugins();
+        final List<String> detachedPlugins = mapToPluginShortName(DetachedPluginsUtil.getDetachedPlugins());
         if (JavaUtils.isRunningWithJava8OrBelow()) {
-            assertEquals(0, detachedPlugins.stream()
-                    .filter(plugin -> plugin.getShortName().equals("jaxb")).count());
-        } else if (JavaUtils.getCurrentJavaRuntimeVersionNumber().isNewerThanOrEqualTo(new VersionNumber("11.0.2"))) {
```

Comment:
this code block was never executed, `JavaUtils.getCurrentJavaRuntimeVersionNumber()` returns the major version of Java, e.g. `11`. I was very confused why it failed on newer versions and not on 11 (as trilead was split in 2.184 and should have caused a test failure).

But turns out that the code wasn't run till after Java 11

fyi @batmat (author of original test)

Reply:
Thanks @timja!  Confirmed by code review of the JavaUtils class and by running the same checks in the Jenkins groovy script console.